### PR TITLE
Feature/mockoon/mockoon#469

### PR DESCRIPTION
**Pull request by barefootdeveloper**
mockoon/mockoon#469

Added new flag to override a stored environments hostname with the CLI.
Added hostname to list display.

### CODE REVIEW

**Comment by 255kb** [comment]
This looks good @barefootdeveloper 
One thing may need changes though. The `--hostname` flag will be available in both `start` and `dockerize` command. I wonder if this make sense at all to let Mockoon server listen on a specific adapter inside the container. I think it would rather be the user's responsibility to use something like this `-p 192.168.x.x:3000:3000` if they want to listen on a specific adapter on the host machine. 
So I would make the following changes:
- make the `--hostname` flag specific to the `start` command
- force Mockoon to listen to `0.0.0.0` when running inside Docker (I think this would need to be added in `./docker/runner.sh` [line 3](https://github.com/mockoon/cli/blob/main/docker/runner.sh#L3): `mockoon-cli start -l 0.0.0.0 "$@"` 
- force Mockoon to listen to `0.0.0.0` when running the Dockerize command by passing a hostname [here](https://github.com/mockoon/cli/blob/main/src/commands/dockerize.ts#L48-L52).

But I am not very comfortable with Docker networking so I may me mistaken. 
Let me know what you think, and I would gladly make the changes. Maybe @mauriceackel you also have an opinion on this?

**Comment by mauriceackel** [comment]
Hi @255kb and @barefootdeveloper here are my tow cents:

As far as I can tell, there will be a new feature in Mockoon that allows users to select the hostname on which to open the server (i.e. no longer only 0.0.0.0). As Guillaume pointed out correctly, this does not make sense inside the docker container so we need to enforce listening on 0.0.0.0 and the user can then map the internal endpoint to whatever hostname-port combo they like (i.e. by using the -p parameter)

However, to enforce it I'm not 100% sure if it would work with setting the `-l` option in the `./docker/runner.sh` as you explained above. Because the user can in theory specify the `-l` parameter again, leading to something like `mockoon-cli start -l 0.0.0.0 -d foo.json -l 192.168.X.X`. From my quick testing of the CLI, it seems as an argument is specified multiple times, the value of the latter one will be used. Hence, we would need to change `mockoon-cli start "$@"` to `mockoon-cli start "$@" -l 0.0.0.0` in `./docker/runner.sh`. However, this would require some testing I guess.

With regards to the docker containers generated by the `dockerize` command, I also agree that the host 0.0.0.0 needs to be enforced. However, I don't think it will be enough to add it to the environment info inside the command execution code as this is completely decoupled from the execution environment. I think one thing we could do is to make the same change as above to the following [line](https://github.com/mockoon/cli/blob/076cdcf8f085c59d126d6b4ae54ace27f18854ae/src/constants/docker.constants.ts#L7). This will result in the host 0.0.0.0 being enforced when the custom docker container is executed.

Finally I also agree that the hostname parameter should be exclusively for the `start` command and not the `dockerize` command.

Cheers, Maurice

**Comment by 255kb** [comment]
Thanks @mauriceackel for your help. Also you are right, your changes are better and make sure any parameter entered by the user are ignored. 
@barefootdeveloper let me know if you need my help on this.

**Comment by 255kb** [comment]
@barefootdeveloper you can also update libs to v2.3.0 both of them, so we can run the CI after you push the changes

**Comment by barefootdeveloper** [comment]
I believe I made all the requested changes. Generating and executing the docker containers worked successfully, but mockoon-cli threw an error because it didn't understand the hostname argument. I believe this is expected since the CLI version that docker would currently install would not have that argument.

**Comment by 255kb** [comment]
> I believe I made all the requested changes. Generating and executing the docker containers worked successfully, but mockoon-cli threw an error because it didn't understand the hostname argument. I believe this is expected since the CLI version that docker would currently install would not have that argument.

Yes, the failure make sense as its installing from NPM where it's not published yet.

File: src/constants/docker.constants.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ RUN npm install -g @mockoon/cli@{{{version}}}
 COPY {{{filePath}}} ./data
 
 # Build runner script
-RUN echo "mockoon-cli start \\"\\$@\\"; sleep infinity & wait \\$!" > mockoon-runner.sh
+RUN echo "mockoon-cli start \\\\"\\\\$@\\\\" -l 0.0.0.0; sleep infinity & wait \\\\$!" > mockoon-runner.sh
```
Comment (255kb) [inline src/constants/docker.constants.ts:7]: Why all the extra `\\`? I fail to see what they are for 🤔

**Pull request merged**